docs/tutorials/client-server.md

```diff
@@ -0,0 +1,230 @@
+- In the `ConnectionString` property, set `Address`: localhost:5557, `Action`: Connect, `Protocol`: TCP.
+
+> [!Note]
+> In Bonsai.ZeroMQ, the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) can have two functions based on its inputs. On its own, as above, the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) operator creates a Dealer socket that listens for messages on the specified network. With the properties specified, we are asking our [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) to listen for messages on the local machine on port 5557 using the TCP protocol. We use the ‘Connect’ argument for the `SocketConnection` property to tell the dealer that it will connect to a static part of the network with a known IP address, in this case the server which we will implement later.
+
+If we add inputs to the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer), it will act as both a sender and receiver of messages on the network. 
+- Before the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) operator add a [`KeyDown`](xref:Bonsai.Windows.Input.KeyDown). 
+- Add a [`String`](xref:Bonsai.Expressions.StringProperty) operator in sequence as input to the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer).
+
+:::workflow
+![Basic Dealer input](~/workflows/dealer-basic-input.bonsai)
+:::
+
+- In the operator properties, set the [`KeyDown`](xref:Bonsai.Windows.Input.KeyDown) `Filter` to the ‘1’ key.
+- Set the [`String`](xref:Bonsai.Expressions.StringProperty) `Value` to ‘Client1’. 
+
+If we run the Bonsai project now, the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) will continue listening for incoming messages on the network, but every time the ‘1’ key is pressed a message containing the string ‘Client1’ will be sent from the socket.
+
+- Copy and paste this client structure a couple of times and change the [`KeyDown`](xref:Bonsai.Windows.Input.KeyDown) and [`String`](xref:Bonsai.Expressions.StringProperty) properties accordingly on each (2, ‘Client2’; 3, ‘Client3’) so that we have 3 total clients that send messages according to different key presses.
+
+:::workflow
+![Multiple clients](~/workflows/multiple-clients.bonsai)
+:::
+
+> [!Note]
+> For the purposes of this article we are creating all of our clients and our server on the same Bonsai project and same machine for ease of demonstration. In a working example, each client and server could be running in separate Bonsai instances on different machines on a network. In this case, localhost would be replaced with the server machine’s IP address.
+
+## Basic server
+Now that we have our client pool set up and sending messages, let’s implement a server to listen for those messages. 
+- Add a [`Router`](xref:Bonsai.ZeroMQ.Router) operator to the project and set its properties to match the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) sockets we already added so that it is running on the same network. 
+- As the [`Router`](xref:Bonsai.ZeroMQ.Router) is acting as server and will be the ‘static’ part of the network, set its `Action` to ‘Bind’.
```

Comment:
It might be worth here to specify exactly the value of the properties to set, even if they are almost identical (except for the `Bind` value). This way people can check their understanding of the parametrization of these socket pairs, i.e. where they are the same, and where they differ.